lib/puppetserver/ca/action/delete.rb

```diff
@@ -192,6 +198,12 @@ def run(args)
             deleted_count += count
           end
 
+          if args['all']
+            certnames = Dir.glob("#{cadir}/signed/*.pem").map{ |c| File.basename(c, '.pem') }
```

Comment:
This will walk the directory twice, once for getting filenames, and then in delete_certs to delete the files. Probably not too expensive.

Reply:
I don't think it's actually walking the directory inside `delete_certs`.  `delete_certs` assumes the file is there and constructs a file path from the certname.  However, you're right in that it might be slightly more efficient to have the function delete the file as it walks the directory.  But I think it's pretty negligible.